_pages/dev/ssr/server-side-rendering-error-handling.md

```diff
@@ -0,0 +1,422 @@
+
+```ts
+@Injectable()
+export class MyService {
+  const errorHandler = inject(ErrorHandler);
+
+  myMethod() {
+    try {
+      // some code that may throw an error
+    } catch (error) {
+      this.errorHandler.handleError(error);
+    }
+  }
+}
+```
+
+## Cache management and error handling
+
+It is not recommended to cache pages for which an error occurred during rendering due to several reasons:
+- *Serving Incorrect Content:* If a page with rendering errors is cached, subsequent requests to that page will receive the erroneous content. This can lead to a poor user experience and confusion, as users might see incomplete or broken pages.
+
+- *Difficulty in Debugging:* When errors are cached, it becomes harder to identify and debug issues. Developers might not be aware that users are receiving erroneous content, making it challenging to diagnose and fix the underlying problem.
+
+- *Performance Degradation:* While caching is generally used to improve performance, caching pages with errors can have the opposite effect. Users might repeatedly encounter the same error, leading to frustration and potentially increasing the load on support channels.
+
+To avoid caching such pages, Spartacus provides changes to the part of `OptimizedSsrEngine`, the `RenderingCache` class, responsible for caching renders. From now, the caching strategy can be controlled by the new `SsrOptimizationOptions` property, `shouldCacheRenderingResult`:
+```ts
+shouldCacheRenderingResult?: ({
+    options,
+    entry,
+  }: {
+    options: SsrOptimizationOptions;
+    entry: Pick<RenderingEntry, 'err' | 'html'>;
+  }) => boolean;
+```
+By default, all html rendering results are cached. By default, also all errors are cached (unless the separate option `avoidCachingErrors` is enabled).
+If needed, the caching strategy can be easily customized by providing an own function.
+
+## Error Handling and Logging
+
+All errors caught during server-side rendering are logged using the `LoggerService` provided by Spartacus, which means they are presented in a standardized way:
+- multiline JSON output in development mode, developer-friendly (TODO: add full stack trace when fixed)
+```ts
+{
+  "message": "OutboundHttpError: Outbound HTTP Error ..."
+  "context": {
+    "timestamp": "2024-09-19T09:19:52.034Z",
+    "request": {
+      "url": "/electronics-spa/en/USD/i-do-not-exist",
+      "uuid": "3432be12-d36b-4f51-946e-69b16cda98b9",
+      "timeReceived": "2024-09-19T09:19:50.128Z"
+    }
+  }
+}
+``
+- single line JSON output in production mode, that are seamlessly parsed by monitoring tools. (TODO: add full stack trace when fixed)
```

Comment:
waiting for [CXSPA-8360](https://jira.tools.sap/browse/CXSPA-8360) and [CXSPA-8385](https://jira.tools.sap/browse/CXSPA-8385)